x/superfluid/keeper/superfluid_asset_store.go

```diff
@@ -27,19 +29,18 @@ func (k Keeper) DeleteSuperfluidAsset(ctx sdk.Context, denom string) {
 	prefixStore.Delete([]byte(denom))
 }
 
-func (k Keeper) GetSuperfluidAsset(ctx sdk.Context, denom string) types.SuperfluidAsset {
+func (k Keeper) GetSuperfluidAsset(ctx sdk.Context, denom string) (types.SuperfluidAsset, error) {
 	asset := types.SuperfluidAsset{}
 	store := ctx.KVStore(k.storeKey)
 	prefixStore := prefix.NewStore(store, types.KeyPrefixSuperfluidAsset)
-	bz := prefixStore.Get([]byte(denom))
-	if bz == nil {
-		return asset
-	}
-	err := proto.Unmarshal(bz, &asset)
+	found, err := osmoutils.Get(prefixStore, []byte(denom), &asset)
 	if err != nil {
-		panic(err)
+		return types.SuperfluidAsset{}, err
+	}
+	if !found {
+		return types.SuperfluidAsset{}, sdkerrors.Wrapf(types.ErrNonSuperfluidAsset, "denom: %s", denom)
```

Comment:
Let's try to avoid sdkerror in the newly added code?

Reply:
I am re-using the currently existing errors in errors.go

I think it would be odd to have two errors that exist for the same error. If we want to move away from sdkerrors, we should do this in a separate refactor.